src/generate.c

```diff
@@ -75,11 +75,17 @@ enable_networkd(const char* generator_dir)
         // LCOV_EXCL_STOP
     }
 
-    g_autofree char* link2 = g_build_path(G_DIR_SEPARATOR_S, generator_dir, "network-online.target.wants", "systemd-networkd-wait-online.service", NULL);
-    _netplan_safe_mkdir_p_dir(link2);
-    if (symlink("/lib/systemd/system/systemd-networkd-wait-online.service", link2) < 0 && errno != EEXIST) {
+    // Mask the generic systemd-networkd-wait-online.service unit, to enforce
+    // "netplan-networkd-wait-online@IFACE.service" wait-online policy
+    g_autofree gchar* masking_link = NULL;
+    masking_link = g_build_path(G_DIR_SEPARATOR_S, rootdir ?: G_DIR_SEPARATOR_S,
+                                "run/systemd/system/systemd-networkd-wait-online.service",
+                                NULL);
+    g_debug("Masking systemd-networkd-wait-online.service via %s -> /dev/null", masking_link);
+    _netplan_safe_mkdir_p_dir(masking_link);
+    if (symlink("/dev/null", masking_link) < 0 && errno != EEXIST) {
         // LCOV_EXCL_START
-        g_fprintf(stderr, "failed to create enablement symlink: %m\n");
+        g_fprintf(stderr, "failed to create masking symlink: %m\n");
         exit(1);
         // LCOV_EXCL_STOP
     }
```

Comment:
note: I'm not sure if that change should be generally included in our upstream codebase. We usually do not want to hard-enforce policy like this, but let everybody do what they want in parallel to Netplan.

Could be an Ubuntu specific patch, though.

Reply:
As discussed elsewhere, I think we should try to instead generate a drop-in like:

```
[Service]
ExecStart=
ExecStart=/lib/systemd/systemd-networkd-wait-online -i eth0 -i wlan0 ... -i ethX
```

where eth0, wlan0, ..., ethx are required interfaces. This would eliminate the need for using the templated units, and hence obviate the need for disabling or masking `systemd-networkd-wait-online.service`.

Reply:
see https://github.com/canonical/netplan/pull/456